File: 4 sem/grafs py + java/dfs.py
def checking(n, graph):
    for i in range(n):
        if not visited[i]:
            dfs(i, n)


def dfs(v, n):
    visited[v] = True
    another_visit[v] = time_checking + 1
    time_checking += 1

    return


f = open("input.txt")
w = open("output.txt", 'w')
n_condition = int(f.readline())
graph_condition = [[] * n_condition for i in range(n_condition)]
for i in range(n_condition):
    line = f.readline().split()
    graph_condition[i] = [int(line[k]) for k in range(n_condition)]
visited = [False] * n_condition
time_checking = 0
another_visit = [0] * n_condition
answer = dfs(n_condition, graph_condition)
for i in range(n_condition):
    w.write(str(answer[i]) + ' ')

File: 4 sem/grafs py + java/Dfs.java
import java.io.*;
import java.util.*;

public class Dfs {
    static int[][] graph;
    static boolean[] used;
    static int counter = 1;
    static ArrayList <Integer> times;

    static void dfs(int v, int n){
        used[v] = true;
        times.set(v, counter++);
        for(int i = 0; i < n; i++)
            if (graph[v][i] == 1 && !used[i])
                dfs(i, n);
    }

    public static void main(String[] args) throws IOException {
        BufferedReader in = new BufferedReader(new FileReader("input.txt"));
        BufferedWriter out = new BufferedWriter(new FileWriter("output.txt"));
        int nCondition = Integer.parseInt(in.readLine());
        graph = new int[nCondition][nCondition];
        used = new boolean[nCondition];
        times = new ArrayList<Integer>();
        for (int i = 0; i < nCondition; i++)
            times.add(0);
        Arrays.fill(used, false);
        for(int i = 0; i < nCondition; i++) {
            StringTokenizer st = new StringTokenizer(in.readLine());
            for (int j = 0; j < nCondition; j++) {
                graph[i][j] = Integer.parseInt(st.nextToken());
            }
        }

        for (int i = 0; i < nCondition; i++)
            if (!used[i])
                dfs(i, nCondition);

        for (int i = 0; i < nCondition; i++)
            out.write(times.get(i).toString() + ' ');
        out.flush();
    }
}